# UFCFVQ-15-M Programming for Data Science
# Programming Task 1

## Student Id: 

### Requirement FR1.1 - Find the arithmetic mean

In [55]:
# add code here
def mean(data_list):
    #length of the list
    n = len(data_list)
    # Mean of the data_list
    mean = sum(data_list) / n
    return mean
#test
my_file = open("task1.dat", "r")   # open the file on mode read
list_content = my_file.readlines()  #read the file to list
list_content = [int(i) for i in list_content] #to convert the list of String to numbers
res = mean(list_content) # calculate the mean
print("The mean of task1.bat is:" ) #print the result
print(res ) #print the result

The mean of task1.bat is:
120.89453125


### Requirement FR1.2 - Find the standard deviation

In [56]:
# add code here
def standard_deviation(list_data):
    # length of the list
    n = len(list_data)
    # Mean of the list
    mean_val = mean(list_data)
    # Calculate the variance
    variance = sum([((x - mean_val) ** 2) for x in list_data]) / n
    deviations = variance ** 0.5
    return deviations
#test
my_file = open("task1.dat", "r")   # open the file on mode read
list_content = my_file.readlines()  #read the file to list
list_content = [int(i) for i in list_content] #to convert the list of String to numbers
res = standard_deviation(list_content) # calculate the standard_deviation
print("The standard deviation of task1.bat is: " ) #print the result
print(res ) #print the result

The standard deviation of task1.bat is: 
31.95179590820272


### Requirement FR1.3 - Find the min/max values 

In [57]:
# add code here
#max of data list
max(data_list)

#min of the data list
min(data_list)

#test

my_file = open("task1.dat", "r")   # open the file on mode read
list_content = my_file.readlines()  #read the file to list
list_content = [int(i) for i in list_content] #to convert the list of String to numbers
res = min(list_content) # calculate the min
print("The Min : " ) #print the result
print(res ) #print the result
res = max(list_content) # calculate the max
print("The Max : " ) #print the result
print(res ) #print the result

The Min : 
0
The Max : 
199


### Requirement FR1.4 - Find the 25th percentile 

In [58]:
# add code here
import math

def percentile_25(data_list):
    size = len(data_list)
    res = sorted(data_list)[int(math.ceil((size * 25) / 100)) - 1]
    return res
#test
my_file = open("task1.dat", "r")   # open the file on mode read
list_content = my_file.readlines()  #read the file to list
list_content = [int(i) for i in list_content] #to convert the list of String to numbers
print("The percentile 25 : " ) #print the result
res = percentile_25(list_content) # calculate the percentile_25
print(res ) #print the result

The percentile 25 : 
99


### Requirement FR1.5 - Find the 50th percentile

In [59]:
# add code here
import math

def percentile_50(data_list):
    size = len(data_list)
    res = sorted(data_list)[int(math.ceil((size * 50) / 100)) - 1]
    return res

#test
my_file = open("task1.dat", "r")   # open the file on mode read
list_content = my_file.readlines()  #read the file to list
list_content = [int(i) for i in list_content] #to convert the list of String to numbers
print("The percentile 50 : " ) #print the result
res = percentile_50(list_content) # calculate the percentile_50
print(res ) #print the result

The percentile 50 : 
117


### Requirement FR1.6 - Find the 75th percentile

In [60]:
# add code here
import math

def percentile_75(data_list):
    size = len(data_list)
    res = sorted(data_list)[int(math.ceil((size * 75) / 100)) - 1]
    return res
#test
my_file = open("task1.dat", "r")   # open the file on mode read
list_content = my_file.readlines()  #read the file to list
list_content = [int(i) for i in list_content] #to convert the list of String to numbers
print("The percentile 75 : " ) #print the result
res = percentile_75(list_content) # calculate the percentile_75
print(res ) #print the result

The percentile 75 : 
140


### Requirement FR1.7 - Print the set of summary statistics in tabular form

In [61]:
# add code here
def summary(data_list):
    print("count", len(data_list) )
    print("mean", mean(data_list) )
    print("std", standard_deviation(data_list) )
    print("min", min(data_list))
    print("25%",percentile_25(data_list) )
    print("50%" ,percentile_50(data_list))
    print("75%", percentile_75(data_list) )
    print("Max", max(data_list) )
#test
my_file = open("task1.dat", "r")   # open the file on mode read
list_content = my_file.readlines()  #read the file to list
list_content = [int(i) for i in list_content] #to convert the list of String to numbers
print("The summary : " ) #print the result
res = summary(list_content) # calculate the mean

The summary : 
count 768
mean 120.89453125
std 31.95179590820272
min 0
25% 99
50% 117
75% 140
Max 199


### Requirement FR1.8 - Read data from a file into memory

In [62]:
# add code here
def read_file(file_name):
    my_file = open(file_name, "r")   # open the file on mode read
    list_content = my_file.readlines()  #read the file to list
    list_content = [int(i) for i in list_content] #to convert the list of String to numbers
    return list_content
#test
list_content = read_file("task1.dat")
list_content = [int(i) for i in list_content] #to convert the list of String to numbers
print(list_content ) #print the result

[148, 85, 183, 89, 137, 116, 78, 115, 197, 125, 110, 168, 139, 189, 166, 100, 118, 107, 103, 115, 126, 99, 196, 119, 143, 125, 147, 97, 145, 117, 109, 158, 88, 92, 122, 103, 138, 102, 90, 111, 180, 133, 106, 171, 159, 180, 146, 71, 103, 105, 103, 101, 88, 176, 150, 73, 187, 100, 146, 105, 84, 133, 44, 141, 114, 99, 109, 109, 95, 146, 100, 139, 126, 129, 79, 0, 62, 95, 131, 112, 113, 74, 83, 101, 137, 110, 106, 100, 136, 107, 80, 123, 81, 134, 142, 144, 92, 71, 93, 122, 163, 151, 125, 81, 85, 126, 96, 144, 83, 95, 171, 155, 89, 76, 160, 146, 124, 78, 97, 99, 162, 111, 107, 132, 113, 88, 120, 118, 117, 105, 173, 122, 170, 84, 96, 125, 100, 93, 129, 105, 128, 106, 108, 108, 154, 102, 57, 106, 147, 90, 136, 114, 156, 153, 188, 152, 99, 109, 88, 163, 151, 102, 114, 100, 131, 104, 148, 120, 110, 111, 102, 134, 87, 79, 75, 179, 85, 129, 143, 130, 87, 119, 0, 73, 141, 194, 181, 128, 109, 139, 111, 123, 159, 135, 85, 158, 105, 107, 109, 148, 113, 138, 108, 99, 103, 111, 196, 162, 96, 184, 81, 1

### Requirement FR1.9 - Design and build a data structure to store CSV data in memory

In [63]:
# add code here
import csv
def read_file_csv(file_name):
    my_file = open(file_name, "r")   # open the file on mode read
    csvFile = csv.reader(my_file, delimiter=',')
  # store the contents of the CSV file in a list of list
    return csvFile

### Requirement FR1.10 - Create and demonstrate a function to read CSV data from a file into memory

In [39]:
# add code here
import csv

with open('task1.csv') as csv_file:
 csv_reader = csv.reader(csv_file, delimiter=',') #the data of CSV will be stored in the csv_reader

print(csv_reader)

### Requirement FR1.11 - Create and demonstrate a generic statistical summary function 

In [64]:
# add code here
import csv
import math
#using the function from FR1.9 to read the file task1.csv
csv_reader = read_file_csv("task1.csv")
line_count = 0
for row in csv_reader:
    if line_count == 0:  # skip the header
        line_count += 1
    else:
        data_list = [float(i) for i in row]
        #using the function from FR1.7 to print the summary
        summary(data_list)

count 9
mean 38.46966666666667
std 45.53401152252383
min 0.0
25% 1.0
50% 33.6
75% 50.0
Max 148.0
count 9
mean 26.55011111111111
std 29.33997576475726
min 0.0
25% 0.351
50% 26.6
75% 31.0
Max 85.0
count 9
mean 34.66355555555556
std 56.17757839627066
min 0.0
25% 0.672
50% 8.0
75% 32.0
Max 183.0
count 9
mean 35.80744444444444
std 35.487212392043766
min 0.0
25% 1.0
50% 23.0
75% 66.0
Max 94.0
count 9
mean 51.043111111111116
std 57.07913825644852
min 0.0
25% 2.288
50% 35.0
75% 43.1
Max 168.0
count 9
mean 27.866777777777777
std 38.76336281071421
min 0.0
25% 0.0
50% 5.0
75% 30.0
Max 116.0
count 9
mean 34.36088888888889
std 30.50141223850479
min 0.248
25% 3.0
50% 31.0
75% 50.0
Max 88.0
count 9
mean 21.048222222222222
std 35.62904406294823
min 0.0
25% 0.0
50% 0.134
75% 29.0
Max 115.0
count 9
mean 104.62866666666667
std 165.26169616298466
min 0.158
25% 2.0
50% 45.0
75% 70.0
Max 543.0
count 9
mean 31.58133333333333
std 45.75979347272742
min 0.0
25% 0.0
50% 1.0
75% 54.0
Max 125.0
count 9
mean 30.421

50% 17.0
75% 27.3
Max 90.0
count 9
mean 58.53322222222222
std 66.02076800148703
min 0.0
25% 1.0
50% 37.4
75% 74.0
Max 204.0
count 9
mean 26.92577777777778
std 37.35835328805408
min 0.0
25% 0.0
50% 4.0
75% 37.0
Max 114.0
count 9
mean 56.943222222222225
std 58.10156692049952
min 1.0
25% 9.0
50% 34.3
75% 86.0
Max 156.0
count 9
mean 91.92077777777779
std 146.53629337454464
min 0.0
25% 1.0
50% 40.6
75% 82.0
Max 485.0
count 9
mean 40.67077777777777
std 58.451707781871384
min 0.0
25% 0.137
50% 8.0
75% 47.9
Max 188.0
count 9
mean 42.037444444444446
std 47.98818483083127
min 0.0
25% 1.0
50% 36.0
75% 50.0
Max 152.0
count 9
mean 34.248555555555555
std 36.60584918382766
min 0.0
25% 2.0
50% 21.0
75% 52.0
Max 99.0
count 9
mean 41.22588888888889
std 46.610166579226146
min 0.0
25% 1.0
50% 23.0
75% 56.0
Max 135.0
count 9
mean 31.91433333333333
std 30.826270167576947
min 0.0
25% 2.0
50% 22.0
75% 53.0
Max 88.0
count 9
mean 55.190777777777775
std 50.75899474045688
min 0.817
25% 17.0
50% 41.0
75% 72.0
Max 

count 9
mean 29.904444444444444
std 30.033820606891055
min 0.0
25% 4.0
50% 20.8
75% 49.0
Max 96.0
count 9
mean 41.373666666666665
std 35.48592291787454
min 0.0
25% 5.0
50% 36.1
75% 72.0
Max 108.0
count 9
mean 32.592666666666666
std 30.75216536549364
min 0.0
25% 0.434
50% 29.0
75% 40.0
Max 88.0
count 9
mean 37.373000000000005
std 35.062583551890945
min 0.0
25% 1.0
50% 30.0
75% 62.0
Max 107.0
count 9
mean 55.94711111111111
std 59.2852567403181
min 1.0
25% 2.0
50% 37.0
75% 78.0
Max 182.0
count 9
mean 53.56811111111111
std 62.13045685123529
min 0.613
25% 1.0
50% 40.5
75% 48.0
Max 194.0
count 9
mean 33.12822222222222
std 50.82471243571221
min 0.0
25% 0.0
50% 0.254
75% 50.0
Max 161.0
count 9
mean 48.24355555555556
std 50.705865308575085
min 0.0
25% 6.0
50% 31.0
75% 62.0
Max 151.0
count 9
mean 74.92633333333333
std 109.7292196110245
min 0.337
25% 2.0
50% 29.0
75% 70.0
Max 360.0
count 9
mean 56.58
std 68.88403524120166
min 0.0
25% 0.52
50% 29.0
75% 84.0
Max 215.0
count 9
mean 53.89022222222222

25% 1.0
50% 22.0
75% 60.0
Max 80.0
count 9
mean 30.08522222222222
std 45.09287024643395
min 0.0
25% 0.0
50% 2.0
75% 29.0
Max 139.0
count 9
mean 25.404222222222224
std 31.408021770021954
min 0.0
25% 1.0
50% 8.0
75% 36.0
Max 90.0
count 9
mean 23.73388888888889
std 44.03423502595438
min 0.0
25% 0.0
50% 0.205
75% 29.0
Max 141.0
count 9
mean 38.73822222222222
std 44.26181046098262
min 0.0
25% 0.244
50% 33.0
75% 41.0
Max 140.0
count 9
mean 31.703777777777777
std 47.01185740209657
min 0.0
25% 0.0
50% 5.0
75% 29.9
Max 147.0
count 9
mean 24.70522222222222
std 33.02209624471259
min 0.0
25% 0.147
50% 15.0
75% 21.0
Max 97.0
count 9
mean 29.94744444444445
std 38.691261818862955
min 0.0
25% 0.0
50% 6.0
75% 36.8
Max 107.0
count 9
mean 43.85944444444445
std 60.284346063027954
min 0.0
25% 0.435
50% 25.0
75% 41.0
Max 189.0
count 9
mean 30.966333333333335
std 28.223045760512807
min 0.0
25% 2.0
50% 23.0
75% 50.0
Max 83.0
count 9
mean 43.27
std 44.41928610162231
min 0.0
25% 4.0
50% 27.0
75% 64.0
Max 120.0


Max 97.0
count 9
mean 31.183222222222224
std 33.58195955429303
min 0.0
25% 1.0
50% 19.5
75% 46.0
Max 100.0
count 9
mean 40.77722222222222
std 41.81564879265437
min 0.0
25% 0.695
50% 27.0
75% 86.0
Max 105.0
count 9
mean 28.95588888888889
std 41.50848681212191
min 0.0
25% 0.0
50% 4.0
75% 34.3
Max 128.0
count 9
mean 34.85311111111111
std 47.707007983790305
min 0.0
25% 0.178
50% 6.0
75% 50.0
Max 147.0
count 9
mean 17.067777777777778
std 28.36192822851623
min 0.0
25% 0.0
50% 0.61
75% 28.0
Max 90.0
count 9
mean 46.14777777777778
std 49.58800970166921
min 0.0
25% 3.0
50% 27.6
75% 72.0
Max 152.0
count 9
mean 86.39266666666667
std 133.43044058484807
min 0.0
25% 2.0
50% 35.0
75% 74.0
Max 440.0
count 9
mean 51.20522222222222
std 60.028684591021936
min 0.447
25% 1.0
50% 23.4
75% 74.0
Max 167.0
count 9
mean 53.91722222222222
std 64.00385869926173
min 0.0
25% 1.0
50% 36.0
75% 50.0
Max 179.0
count 9
mean 51.95111111111111
std 49.52406383532587
min 0.26
25% 11.0
50% 35.0
75% 84.0
Max 136.0
count 9
mea

### Requirement FR1.12 - Create and demonstrate a function to print a custom table

In [68]:
# add code here
import csv

with open('task1.csv') as csv_file:
    csv_reader = csv.reader(csv_file, delimiter=',')
    line_count = 0
    for row in csv_reader:
        if line_count == 0: #skip the header
            line_count += 1
        else:
            print("####################################################") #just separator
            row = [float(i) for i in row]
            summary(row) #call the function from FR1.7

####################################################
count 9
mean 38.46966666666667
std 45.53401152252383
min 0.0
25% 1.0
50% 33.6
75% 50.0
Max 148.0
####################################################
count 9
mean 26.55011111111111
std 29.33997576475726
min 0.0
25% 0.351
50% 26.6
75% 31.0
Max 85.0
####################################################
count 9
mean 34.66355555555556
std 56.17757839627066
min 0.0
25% 0.672
50% 8.0
75% 32.0
Max 183.0
####################################################
count 9
mean 35.80744444444444
std 35.487212392043766
min 0.0
25% 1.0
50% 23.0
75% 66.0
Max 94.0
####################################################
count 9
mean 51.043111111111116
std 57.07913825644852
min 0.0
25% 2.288
50% 35.0
75% 43.1
Max 168.0
####################################################
count 9
mean 27.866777777777777
std 38.76336281071421
min 0.0
25% 0.0
50% 5.0
75% 30.0
Max 116.0
####################################################
count 9
mean 34.36088888888889
std 30.5014

####################################################
count 9
mean 33.26166666666667
std 38.10793851388157
min 0.0
25% 0.855
50% 30.0
75% 38.0
Max 109.0
####################################################
count 9
mean 33.39388888888888
std 40.8963971367336
min 0.0
25% 0.0
50% 2.0
75% 54.0
Max 109.0
####################################################
count 9
mean 28.65933333333333
std 30.93371959528954
min 0.0
25% 1.0
50% 19.6
75% 38.0
Max 95.0
####################################################
count 9
mean 46.45433333333333
std 48.79458070091164
min 0.0
25% 4.0
50% 27.0
75% 85.0
Max 146.0
####################################################
count 9
mean 37.863
std 36.24867931387294
min 0.867
25% 2.0
50% 28.0
75% 66.0
Max 100.0
####################################################
count 9
mean 48.66788888888889
std 52.11531896870514
min 0.0
25% 5.0
50% 28.6
75% 64.0
Max 140.0
####################################################
count 9
mean 35.10922222222222
std 43.07613740492983
min 

mean 34.248555555555555
std 36.60584918382766
min 0.0
25% 2.0
50% 21.0
75% 52.0
Max 99.0
####################################################
count 9
mean 41.22588888888889
std 46.610166579226146
min 0.0
25% 1.0
50% 23.0
75% 56.0
Max 135.0
####################################################
count 9
mean 31.91433333333333
std 30.826270167576947
min 0.0
25% 2.0
50% 22.0
75% 53.0
Max 88.0
####################################################
count 9
mean 55.190777777777775
std 50.75899474045688
min 0.817
25% 17.0
50% 41.0
75% 72.0
Max 163.0
####################################################
count 9
mean 38.77711111111111
std 48.31171622436125
min 0.0
25% 0.294
50% 29.7
75% 38.0
Max 151.0
####################################################
count 9
mean 45.60044444444444
std 38.36705301604233
min 0.0
25% 7.0
50% 40.0
75% 74.0
Max 105.0
####################################################
count 9
mean 64.92966666666668
std 85.96957689787708
min 0.0
25% 0.167
50% 34.0
75% 80.0
Max 285.0
##

mean 51.96244444444444
std 55.62769888605878
min 0.0
25% 7.0
50% 32.4
75% 88.0
Max 155.0
####################################################
count 9
mean 36.95166666666667
std 51.86666259640087
min 0.0
25% 0.165
50% 7.0
75% 47.0
Max 161.0
####################################################
count 9
mean 30.02877777777778
std 38.566709345345956
min 0.0
25% 0.259
50% 2.0
75% 52.0
Max 108.0
####################################################
count 9
mean 50.62744444444444
std 50.69645553052032
min 0.0
25% 7.0
50% 26.0
75% 74.0
Max 136.0
####################################################
count 9
mean 100.70211111111112
std 163.79283545615002
min 0.0
25% 5.0
50% 38.7
75% 84.0
Max 545.0
####################################################
count 9
mean 60.156444444444446
std 68.17610577697717
min 0.808
25% 1.0
50% 39.0
75% 86.0
Max 220.0
####################################################
count 9
mean 29.904444444444444
std 30.033820606891055
min 0.0
25% 4.0
50% 20.8
75% 49.0
Max 96.0
##

####################################################
count 9
mean 38.95144444444445
std 40.237750586320224
min 0.0
25% 4.0
50% 22.1
75% 72.0
Max 116.0
####################################################
count 9
mean 33.967
std 50.24964456259036
min 0.0
25% 0.803
50% 4.0
75% 32.9
Max 158.0
####################################################
count 9
mean 60.03333333333334
std 84.75903098392132
min 0.0
25% 2.0
50% 25.0
75% 58.0
Max 275.0
####################################################
count 9
mean 40.96044444444445
std 39.12112111297173
min 0.0
25% 3.0
50% 34.0
75% 56.0
Max 115.0
####################################################
count 9
mean 32.72177777777778
std 41.017741077023345
min 0.0
25% 0.196
50% 22.0
75% 40.0
Max 131.0
####################################################
count 9
mean 22.387666666666668
std 30.170796682296018
min 0.0
25% 0.0
50% 3.0
75% 25.0
Max 82.0
####################################################
count 9
mean 39.79344444444444
std 58.415564986494466

std 36.45817934138908
min 0.0
25% 0.0
50% 5.0
75% 30.0
Max 110.0
####################################################
count 9
mean 33.516333333333336
std 31.9221410587977
min 0.0
25% 2.0
50% 25.0
75% 72.0
Max 81.0
####################################################
count 9
mean 59.029222222222224
std 64.4498037955427
min 0.163
25% 7.0
50% 33.0
75% 70.0
Max 195.0
####################################################
count 9
mean 58.68211111111111
std 65.7342143626631
min 0.0
25% 6.0
50% 32.0
75% 74.0
Max 193.0
####################################################
count 9
mean 38.39033333333333
std 40.834052684820136
min 0.0
25% 2.0
50% 21.0
75% 71.0
Max 117.0
####################################################
count 9
mean 28.496333333333332
std 30.06020286765285
min 0.0
25% 0.267
50% 28.0
75% 37.2
Max 84.0
####################################################
count 9
mean 21.858555555555554
std 24.13931894054967
min 0.0
25% 0.727
50% 6.0
75% 41.0
Max 68.0
###############################

50% 2.0
75% 30.8
Max 108.0
####################################################
count 9
mean 38.89577777777778
std 32.713380347013896
min 0.0
25% 4.0
50% 37.7
75% 54.0
Max 90.0
####################################################
count 9
mean 26.545111111111108
std 40.76810589022432
min 0.0
25% 0.0
50% 0.206
75% 24.7
Max 125.0
####################################################
count 9
mean 29.310333333333332
std 43.801387202791744
min 0.0
25% 0.0
50% 0.393
75% 32.4
Max 132.0
####################################################
count 9
mean 32.52711111111111
std 42.82920099507512
min 0.0
25% 0.0
50% 5.0
75% 45.0
Max 128.0
####################################################
count 9
mean 25.649777777777775
std 31.075020342454337
min 0.0
25% 0.148
50% 21.0
75% 24.7
Max 94.0
####################################################
count 9
mean 27.570222222222224
std 36.83030371183725
min 0.0
25% 0.732
50% 7.0
75% 34.0
Max 114.0
####################################################
count 9
mea

std 121.99397512992189
min 0.826
25% 7.0
50% 33.9
75% 50.0
Max 392.0
####################################################
count 9
mean 60.53
std 67.41121963985915
min 0.97
25% 3.0
50% 33.8
75% 78.0
Max 185.0
####################################################
count 9
mean 30.410555555555554
std 32.982734578062406
min 0.0
25% 0.595
50% 18.0
75% 56.0
Max 94.0
####################################################
count 9
mean 50.32388888888889
std 55.88577446590788
min 0.0
25% 1.0
50% 35.5
75% 60.0
Max 178.0
####################################################
count 9
mean 32.664222222222214
std 33.844317040688196
min 0.0
25% 1.0
50% 27.0
75% 52.0
Max 97.0
####################################################
count 9
mean 28.40188888888889
std 32.34636310809899
min 0.0
25% 0.317
50% 19.0
75% 34.0
Max 86.0
####################################################
count 9
mean 52.93211111111111
std 62.65019113289005
min 0.0
25% 1.0
50% 36.0
75% 66.0
Max 200.0
#####################################

# Process Development Report for Task 1


Report for Programming Task 1
The most common language that is used in programming is Python, as it has lots of useful functions built within that helps in performing studies on big data and analyzing the provided data. In this assignment, I have worked on several functions to achieve the requirements as shown and mentioned below.
•	F1.1:
The mean is the average of the numbers, in this task, we assume we have a list of data stored in memory and then calculate the mean.
To test the same, we read the file task1.bat and calculated the mean through a built-in module function

•	F1.2:
IN this task we are required to calculate the standard deviation, and the formula is:
σ=√1N∑Ni=1(Xi−μ)2 σ = 1 N ∑ i = 1 N ( X i − μ ) 2.

To test the same, we read the file task1.bat and calculated the SD as per the formula to find the SD of the sample data accordingly, instead of using the NumPy function

•	F1.3:
The min/max values were calculated from the Math package that is provided by Python. The minimum value is 0 as per the calculation and the maximum value is 199.

•	F1.4 to F1.6:
Percentiles are used in statistics to give you a number that describes the value that a given percent of the values are lower than.
The Formula is: n = (P/100) x N
where N = number of values in the data set, P = percentile, and n = ordinal rank of a given value. The 25th percentile is 99, the 50th percentile is 117, and the 75th percentile is 140.

•	F1.7 to F1.12:
The summary function is to display the previous function by passing a list of data from memory. The total count of the data is 768, while the mean of the data used is 120.89, and the standard deviation is 31.95.
According to the percentile part, the 25th, 50th, and 75th were calculated. The answer is as following 99, 117, and 140.
Statistically, the minimum and the maximum values were also calculated, whereas the min value is 0, and the max value is 199.
Task1.dat file initially contained a single data value, but then replaced as per the instruction as list data structure format. The data structure is capable to store multiple data values per CSV row, this was done by importing CSV, then using the def “read_file_csv” function, and then storing it by using “return CSV file”. In the last task, a function was developed to generate statistically the output in a table form.

In conclusion, a data scientist needs to perform such analysis to get the benefit of the basic functions that are provided in Python. Moreover, this task made it easier for me to read big data and start analyzing them properly and more analytically.

